### PR TITLE
Close fyst #187355990

### PR DESCRIPTION
**Pull request by tofarr**
* Lots of copy updates in here
* Added before_action to redirect to either the return-status or the welcome page depending on whether a submission is present for the intake.
* Skipping the redirect for select pages
* Using sessions_toggle for checking dates (Making testing easier).
* Using different config for testing to prevent test failures after the closing day

### CODE REVIEW

**Comment by tofarr** [comment]
Co-authored-by: Em Barnard-Shao <ebarnard@codeforamerica.org>

File: app/assets/stylesheets/_state-file.scss [hunk 1]
```diff
@@ -673,13 +673,10 @@
       max-width: 100%;
     }
 
-    .esign-declaration-outer, .return-status-outer, .submission-confirmation-outer {
-
-      .question-wrapper .button--primary {
-        min-width: 40rem;
-        max-width: 40rem;
-        width: 40rem;
-      }
```
Comment (tofarr) [inline app/assets/stylesheets/_state-file.scss:682]: I noticed some text overflow on some buttons - making this a more generic rule allowed it to be reused. (Particularly in the different states of the return-status page)

File: app/assets/stylesheets/_state-file.scss [hunk 2]
```diff
@@ -673,13 +673,10 @@
...
+    .question-layout .question-wrapper .button--primary.button--wide {
+      min-width: 40rem;
+      max-width: 40rem;
+      width: 40rem;
```
Comment (tofarr) [inline app/assets/stylesheets/_state-file.scss:679]: I noticed some text overflow on some buttons - making this a more generic rule allowed it to be reused. (Particularly in the different states of the return-status page)

File: config/environments/test.rb [hunk 3]
```diff
@@ -106,6 +106,8 @@
 
   # StateFile
   config.state_file_start_of_open_intake = Time.find_zone('America/New_York').parse('2024-01-01 7:59:59')
+  config.state_file_end_of_new_intakes = Time.find_zone('America/New_York').parse('2038-04-15 23:59:59')
+  config.state_file_end_of_in_progress_intakes = Time.find_zone('America/New_York').parse('2038-04-25 23:59:59')
```
Comment (tofarr) [inline config/environments/test.rb:110]: Setting this means that tests will not start failing on April 15

File: app/views/state_file/questions/return_status/_rejected.html.erb [hunk 4]
```diff
@@ -39,20 +39,26 @@
...
 
 <% if @error&.auto_wait %>
-  <section class="spacing-below-15">
-    <p class="spacing-below-5"><strong><%= t('.next_steps.can_edit.title') %></strong></p>
-    <%= @error.resolution(I18n.locale).present? ? @error.resolution(I18n.locale) : t('.next_steps.can_edit.body') %>
-  </section>
+  <% if app_time.before?(Rails.configuration.state_file_end_of_in_progress_intakes) %>
+    <section class="spacing-below-15">
+      <p class="spacing-below-5"><strong><%= t('.next_steps.can_edit.title') %></strong></p>
+      <%= @error.resolution(I18n.locale).present? ? @error.resolution(I18n.locale) : t('.next_steps.can_edit.body') %>
```
Comment (embarnard) [inline app/views/state_file/questions/return_status/_rejected.html.erb:45]: @tofarr from what tiffany said in slack I think we actually need to show the default text if its after this time so this `<%= @error.resolution(I18n.locale).present? ? @error.resolution(I18n.locale) : t('.next_steps.can_edit.body') %>` should actually be something more like this `<%= @error.resolution(I18n.locale).present? && app_time.before?(Rails.configuration.state_file_end_of_in_progress_intakes) ? @error.resolution(I18n.locale) : t('.next_steps.can_edit.body') %>`

File: app/views/state_file/questions/return_status/_rejected.html.erb [hunk 5]
```diff
@@ -39,20 +39,26 @@
 <% end %>
```
[the remark this answers is not in the document]
Comment (embarnard) [reply app/views/state_file/questions/return_status/_rejected.html.erb:39]: line 37 too

File: spec/controllers/state_file/questions/return_status_controller_spec.rb [hunk 6]
```diff
@@ -0,0 +1,18 @@
+require "rails_helper"
+
+RSpec.describe StateFile::ReturnStatusController, type: :controller do
+  #TODO : set up the intake
```
Comment (embarnard) [inline spec/controllers/state_file/questions/return_status_controller_spec.rb:4]: remove this todo?

**Pull request merged**